src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -499,6 +519,10 @@ class GroupReadsByUmi
 
   private val assigner = strategy.newStrategy(this.edits)
 
+  // Give values to unset parameters that are different in duplicate marking mode
```

Comment:
Coupling these seems odd to me & deviates from the intended change set.

Reply:
Can you provide a bit more context of why it seems odd?  My intuition is that if you're marking duplicates with this tool, you want to keep all reads (i.e. don't pre-filter based on mapq).  Similarly, you want to set the duplicate flag on secondary and supplementary reads (and don't prefilter them) too.  What am I missing?

Also, what's the "intended change set"?

Reply:
Given that the tool allows for multiple versions of filtering, allowing flexibility to choose what one is filtering & not seems like it's giving appropriate control over the functionality, rather than forcing a workflow. 

It's true that, if one is keeping all reads, one must update all reads, but forcing all reads to stay, when marking doesn't seem like a requirement for that.

Reply:
I don't think any option is "forced", it just has a default when the option is unset.  You can always set the option and this will override the default.  For example, you can explicitly set `--include-suppementary=true` or `--include-suppementary=false`.  I think that's why the types on the command line parameters are `Option[Boolean]` with default `None`:

1. `None` -> use a default
2. `Some(true)` -> include supplementary
3. `Some(false)` -> filter out supplementary

Reply:
Fair. I'm being overly dramatic with "forced" & you're right, it can be overidden, you're the maintainers so the paradigm the tool uses is ultimately up to you. 

I just strikes me as odd to change defaults when semi related parameters are passed - they aren't strictly coupled, parameters, if they were strictly coupled i'd agree - i think this makes for a usage that implies a specific workflow & causes users to have to more closely read the docs if their workflow isn't that one. 

As a user, when defaults change because of other parameters it can result in unexpected results - which are often hard to debug.

Reply:
Fair, the alternative is to provide a second command line tool that performs the duplicate marking that is just a facade over group reads by umi, but that seems a bit heavy handed.  I'll let @tfenne take this discussion over, as I don't have strong feelings either way, but would lean to the current implementation as it is well-documented in the usage.

Reply:
I made this change because otherwise, if one wants to use `GroupReadsByUmi` as an equivalent to `MarkDuplicates` in picard, one would have had to add:

`--mark-duplicates --min-mapq 0 --include-secondary --include-supplementary`

which to me seems error prone.  I don't see the harm in defaulting arguments one way when the intended usage is for consensus calling and another when the intended usage is for duplicate marking.  The user is always free to set them explicitly.